### PR TITLE
License namespace model

### PR DESCRIPTION
**Pull request by tjasmith**
I have added the following:
- license namespace model
- organisation model
- updated the navbar to accomodate a new menu
- Added the submit license namespace form, and template

### CODE REVIEW

File: src/src/secret.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 def getGithubKey():
-    return 'GHKEYXXX'
+    return '376d679b554ae9a9d474'
```
Comment (techytushar) [inline src/src/secret.py]: Be careful and don't commit your GitHub App ID and Secret Key, review your changes before pushing them

File: src/src/secret.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 
 def getGithubSecret():
-    return 'GHSECRETXXX'
+    return 'b0b172b5780173783a4c86d60bae983c4608d6a7'
```
Comment (techytushar) [inline src/src/secret.py]: Same here!!

**Comment by techytushar** [comment]
There is still work left to be done, leave a comment when you think its ready for review, nice work so far :+1:

File: src/src/secret.py (hunk 2 above)
Comment (tjasmith) [reply src/src/secret.py]: Thanks for pointing that out. I have corrected this.

File: src/src/secret.py (hunk 1 above)
Comment (tjasmith) [reply src/src/secret.py]: Thanks for pointing that out. I have corrected this.

**Comment by tjasmith** [comment]
@techytushar Thanks,
I pushed this so you all can see my progress and suggest changes which I can implement as I progress.

**Comment by goneall** [comment]
Looks good so far - I like the dropdown for the license requests - solves some horizontal real estate issues.

**Comment by tjasmith** [comment]
@techytushar @goneall 

- I have added a method to verify whether a license is already present on the spdx license list
- added the possibility to create an organisation(via the admin ui)
- added namespace id to the model and form
- added the publiclyShared boolean to the form

File: src/app/utils.py [hunk 3]
```diff
@@ -203,3 +205,54 @@ def check_license_name(name):
...
+    license_list = getLicenseList(token)
+    return_dict = {
+    "name": "",
+    "licenseId": "",
+    "referenceNumber": "",
+    "isDeprecatedLicenseId": "",
+    "exists": False
+    }
+    for license in license_list["licenses"]:
+        if namespaceId == license["licenseId"] or namespace == license["name"]:
```
Comment (goneall) [inline src/app/utils.py]: You can also compare the license text to see if it matches the listed licenses.  See https://github.com/spdx/spdx-online-tools/blob/ecd4426e8ccdd5ecd79fae10ede31142f900f77a/src/app/views.py#L880 for the code to do this.

Note that  this method is currently rather slow.  I'm planning on making some changes to speed this up over the next month or so.

**Comment by techytushar** [comment]
@tjasmith you branch has conflicts with the `master` branch, please update your branch with master, resolve the conflicts and push again.

**Comment by goneall** [comment]
Looking good so far - @tjasmith @techytushar Let me know if you want me to merge any code into master

File: src/app/utils.py [hunk 4]
```diff
@@ -203,3 +205,53 @@ def check_license_name(name):
...
+    response = requests.get(license_list_url, headers=headers)
+    data = json.loads(response.text)
+    return data
+
+
+def licenseInList(namespace, namespaceId, token):
+    license_list = getLicenseList(token)
+    return_dict = {
+    "name": "",
+    "licenseId": "",
```
Comment (techytushar) [inline src/app/utils.py]: There is no need to define all the keys in the `return_dict` beforehand, instead you can only add the `exists` keys to the dict and the rest will be added from lines `243 . . .` if the license matches.

File: src/app/views.py [hunk 5]
```diff
@@ -202,10 +310,23 @@ def createIssue(licenseAuthorName, licenseName, licenseIdentifier, licenseCommen
     title = 'New license request: ' + licenseIdentifier + ' [SPDX-Online-Tools]'
     payload = {'title' : title, 'body': body, 'labels': ['new license/exception request']}
     headers = {'Authorization': 'token ' + token}
-    url = TYPE_TO_URL[urlType]
+    url = TYPE_TO_URL_LICENSE[urlType]
+    r = post(url, data=dumps(payload), headers=headers)
+    return r.status_code
+
+
+def createLicenseNamespaceIssue(licenseNamespace, token, urlType):
```
Comment (techytushar) [inline src/app/views.py]: This function can be defined in the `utils` file and then imported in the `views` file

File: src/app/utils.py (hunk 4 above)
Comment (tjasmith) [reply src/app/utils.py]: @techytushar You want me to make the URL field required?

**Comment by tjasmith** [comment]
@techytushar I have corrected the code as you mentionned.

File: src/app/utils.py (hunk 4 above)
Comment (techytushar) [reply src/app/utils.py]: No, to apply a check on the URL field, so that user enters a valid URL, as it happens in the e-mail field.

Comment (tjasmith) [reply src/app/utils.py]: Ok, I have done that.

**Comment by tjasmith** [comment]
@techytushar @goneall Please could you have a look at this?

**Comment by goneall** [comment]
I noticed that there is a merge conflict with src/app/review - @tjasmith if you could resolve the merge conflict, I'll go ahead and merge.

**Comment by tjasmith** [comment]
@goneall Thanks for pointing that out. I have resolved the merge conflict.

**Pull request merged**